### PR TITLE
builtin/aws/ecs: add logic to determine Deployment.LoadBalancerArn 

### LINKED ISSUE #732: waypoint up with alb.listener_arn doesn't complete
**Describe the bug**
When I run `waypoint up` with an alb.listener_arn set, the ECS service says Created, but it's not attached to the Target group, and reports that the transport is closing.

```
» Deploying...
✓ Found existing ECS cluster: test-cluster
✓ Found existing IAM role to use: test-task-role
✓ Created ALB target group
✓ Modified ALB Listener to introduce target group
✓ Configured security group: ruby-test-inbound-internal
✓ Created ECS Service (ruby-test-T2Z7E5EV9BAD6R24EK2QWQ, cluster-name: test-cluster)
! transport is closing
```

**Steps to Reproduce**
```
project = "ruby-test"

app "ruby-test" {
  labels = {
    "service" = "ruby-test",
    "env" = "dev"
  }

  build {
    use "pack" {}
    registry {
      use "aws-ecr" {
        region = "us-west-1"
        repository = "tests"
        tag = "latest"
      }
    }
  }

  deploy {
    use "aws-ecs" {
      region = "us-west-1"
      memory = "512"
      cluster = "test-cluster"
      # provided by Ops
      log_group = "ruby-test"
      role_name = "test-task-role"
      alb {
        listener_arn = "arn:aws:elasticloadbalancing:us-west-1:123456789101:listener/app/ruby-test/93b4510da291c4cb/b79c7e5fc37f9c8a"
        domain_name = ""
        zone_id = ""
        certificate = ""
      }
      subnets = [
        "subnet-afef18a51f44bb9b2",
        "subnet-e1e1611bfa8e27f45"
      ]
    }
  }
}
```

**Expected behavior**
That the deploy will work properly

### PR DESCRIPTION
**Pull request by anGie44**
Closes #732 

(nil pointer) Behavior reproduced before code change:

1. `waypoint`
```
» Deploying...
✓ Found existing ECS cluster: waypoint
✓ Found existing IAM role to use: ecr-example-nodejs
✓ Created ALB target group
✓ Modified ALB Listener to introduce target group
✓ Configured security group: example-nodejs-inbound-internal
✓ Created ECS Service (example-nodejs-4RMW0A38SAH0K55VV, cluster-name: waypoint)
! transport is closing
```
2. `$ aws elbv2 describe-target-groups --target-group-arns TARGET_GROUP_ARN`
```
{
    "TargetGroups": [
        {
            "TargetGroupArn": "arn:aws:elasticloadbalancing:us-west-2:xxxxxxxxxxx:targetgroup/example-nodejs-4RMW0A38SAH0K55VV/6044a6a52d1d7b11",
            "TargetGroupName": "example-nodejs-4RMW0A38SAH0K55VV",
           ...
            "LoadBalancerArns": [],
            "TargetType": "ip",
            "ProtocolVersion": "HTTP1"
        }
    ]
}
```

Behavior after code change:
```
» Deploying...
✓ Found existing ECS cluster: waypoint
✓ Found existing IAM role to use: ecr-example-nodejs
✓ Created ALB target group
✓ Modified ALB Listener to introduce target group
✓ Configured security group: example-nodejs-inbound-internal
✓ Created ECS Service (example-nodejs-0316CDR3RECCXMSAW, cluster-name: waypoint)

» Releasing...

» Pruning old deployments...
  Deployment: 01ERBEWXF3YZEVVM6P9442K0S4
  Deployment: 01ERBCPEQJ8E92TZB5FEJHDF0M

The deploy was successful! A Waypoint deployment URL is shown below. This
can be used internally to check your deployment and is not meant for external
traffic. You can manage this hostname using "waypoint hostname."
```
2. `$ aws elbv2 describe-target-groups --target-group-arns TARGET_GROUP_ARN`
```
{
    "TargetGroups": [
        {
            "TargetGroupArn": "arn:aws:elasticloadbalancing:us-west-2:xxxxxxx:targetgroup/example-nodejs-0316CDR3RECCXMSAW/fdef3f089a2fd67f",
            "TargetGroupName": "example-nodejs-0316CDR3RECCXMSAW",
           ...
            "LoadBalancerArns": [
                "arn:aws:elasticloadbalancing:us-west-2:xxxxxxx:loadbalancer/app/waypoint-ecs-example-nodejs/de4278aee3d284da"
            ],
        }
    ]
}
```

### CODE REVIEW

File: builtin/aws/ecs/platform.go [hunk 1]
```diff
@@ -1105,11 +1105,21 @@ func (p *Platform) Launch(
...
+	// For deployments without a pre-existing ALB Listener,
+	// we set the LoadBalancerArn directly from the lb found or created.
+	// While for deployments with a pre-existing ALB Listener,
+	// we determine the associated LoadBalancer from the listener
+	// returned from the elbv2.DescribeListeners API response
+	if newListener {
+		dep.LoadBalancerArn = aws.StringValue(lb.LoadBalancerArn)
+	} else {
+		dep.LoadBalancerArn = aws.StringValue(listener.LoadBalancerArn)
 	}
```
Comment (anGie44) [inline builtin/aws/ecs/platform.go:1123]: looking this over again, it seems in the case of no pre-existing ALB listener, the *listener pointer gets set in that branch of the if/else logic so we could maybe simplify this to 
```suggestion
		Cluster:        clusterName,
		TaskArn:        taskArn,
		ServiceArn:     *servOut.Service.ServiceArn,
		TargetGroupArn: *tgArn,
		LoadBalancerArn: aws.StringValue(listener.LoadBalancerArn),
	}
```
🤔 ?

File: builtin/aws/ecs/platform.go [hunk 2]
```diff
@@ -970,10 +975,6 @@ func (p *Platform) Launch(
 			}
 		}
 
-		if len(tgs) == 0 {
```
Comment (anGie44) [inline builtin/aws/ecs/platform.go:973]: should be safe to remove as `tgs` is instantiated w/one TargetGroupTuple @ L836

File: builtin/aws/ecs/platform.go [hunk 3]
```diff
@@ -961,6 +961,11 @@ func (p *Platform) Launch(
 	if !newListener {
 		def := listener.DefaultActions
 
+		// Direct all traffic to the new target group;
+		// effectively, removing traffic from any pre-existing
+		// forwarding default actions
+		tgs[0].Weight = aws.Int64(100)
```
Comment (anGie44) [inline builtin/aws/ecs/platform.go:967]: not too sure what the expected UX should be here 😅 if a new target group is created for the purpose of the deployment, would make sense to send traffic its way..however, how much, and how should it affect already configured target groups in the ALB listener?

Comment (dlundgren) [reply builtin/aws/ecs/platform.go:967]: I noticed the same thing, this causes the need to double deploy, and I had the idea to use the Releaser in #794

**Review by krantzinator — CHANGES_REQUESTED** [review]
Thanks for this! I answered your questions and suggested including a comment so the intent here is more clear for future folks.

**Pull request closed without merging**